scipy/signal/windows/_windows.py

```diff
@@ -37,7 +41,17 @@ def _truncate(w, needed):
         return w
 
 
-def general_cosine(M, a, sym=True):
+def _asarray(x, xp, *, device=None):
+    """A shim for the `device` arg of `np.asarray(x, device=device)` not recognized
+       in NumPy < 2.
+    """
+    if xp == np or is_cupy(xp):
+        return xp.asarray(x)
+    else:
+        return xp.asarray(x, device=device)
```

Comment:
The problem here is that `xp` is user input, so the pattern is

```
def taylor(...., xp=None, device=None):
     xp = np_compat if xp is None else xp
     ....
     xp_acosh(..., xp=xp)
    ...
```

So what is your suggestion?

Reply:
`xp = array_namespace(xp.empty(0))`, or some other function that we can reasonably expect any array library to have.

Reply:
Since `taylor` does not take any array inputs should it not be the case that the user just calls `xp.asarray` on the output like they have to for the `datasets` submodule

Reply:
But `taylor` still uses functions from `xp` to create the output. The point is that a CuPy user doesn't want all computation done on the CPU, if they can instead pass `xp=cp`.

Reply:
> xp = array_namespace(xp.empty(0)), or some other function that we can reasonably expect any array library to have.

IMO it does not matter all that much as long as the shims are temporary: we'll be able to drop them once we stop supporting numpy < 2 (BTW, when will that be?) and cupy < 14. 

I can move the shims to `signal/_windows.py` so that they are less annoying and less inviting, as it were.

Reply:
If it's that big of  a deal, sure; done in https://github.com/scipy/scipy/pull/21783/commits/4558565baea957213e5f3607dddb8f1918208a35

Reply:
thanks!